src/Agent/NewRelic/Agent/Core/JsonConverters/LogEventWireModelCollectionJsonConverter.cs

```diff
@@ -109,7 +109,16 @@ private static void WriteJsonImpl(JsonWriter jsonWriter, LogEventWireModelCollec
                     foreach (var item in logEvent.ContextData)
                     {
                         jsonWriter.WritePropertyName(Context + "." + item.Key);
-                        jsonWriter.WriteValue(item.Value);
+                        string contextValueJson;
+                        try
+                        {
+                            contextValueJson = JsonConvert.SerializeObject(item.Value);
+                        }
+                        catch
+                        {
+                            contextValueJson = item.Value.GetType().ToString();
```

Comment:
Did you mean to use `GetType()` here?

Reply:
Yes, my thinking is that if we can get valid JSON from JsonConvert.SerializeObject for whatever the value object is, as a fallback just send the string representation of the object's type.  I suppose that's what `Object.ToString()` does anyway if there's no override of ToString() for that object, so maybe we could skip the GetType() altogether.

Reply:
Yeah, that's kind of what I was thinking. It's possible they've implemented `ToString()` which would be more useful than the type. But if we can't serialize it, it'll most likely end up just being the type.